library/xml-model/src/test/java/org/cru/godtools/xml/model/ManifestTest.kt

```diff
@@ -75,6 +75,12 @@ class ManifestParsingRobolectricTest {
         assertEquals(0, manifest.tips.size)
     }
 
+    @Test
+    fun testCardBackgroundColorFallbackBehavior() {
+        val manifest = parseManifest("manifest_tract.xml")
```

Comment:
for testing the fallback behavior I want to avoid parsing xml, so create your objects like this instead:

```kotlin
val manifest = Manifest(backgroundColor = Color.GREEN)
```

Reply:
This logic should cover testing both fallback scenarios:

```kotlin
assertEquals(Color.GREEN, Manifest(backgroundColor = Color.RED, cardBackgroundColor = Color.GREEN))
assertEquals(Color.GREEN, Manifest(backgroundColor = Color.GREEN))
```

Reply:
you will probably need to add `backgroundColor` and `cardBackgroundColor` to the Manifest test constructor

Reply:
Oh yeah, will do!

Reply:
So ```backgroundColor``` doesn't have a private val ```_backgroundColor``` like ```cardBackgroundColor``` so trying to add ```backgroundColor``` to the test constructor is throwing errors. So I changed it to this
```
@ColorInt
private val _backgroundColor: Int?
@get:ColorInt
val backgroundColor: Int get() = _backgroundColor ?: DEFAULT_BACKGROUND_COLOR
```

Is this correct or should I just make the test constructor argument be ```_backgroundColor```?

Reply:
neither, you will just need to prefix where you assign it with `this.backgroundColor`

Reply:
take a look at how `code` and `locale` are set in the test constructor

Reply:
Ah cool!